src/librustdoc/passes/collect_intra_doc_links/early.rs

```diff
@@ -181,6 +239,9 @@ impl IntraLinkCrateLoader<'_, '_> {
         for child in self.resolver.module_children_or_reexports(module_id) {
             if child.vis == Visibility::Public {
                 if let Some(def_id) = child.res.opt_def_id() {
+                    if !def_id.is_local() {
+                        self.load_links_in_attrs_extern(def_id, def_id);
```

Comment:
maybe move this check inside `load_links_in_attrs_extern` so it's done consistently?